src/builtins/roc/Result.roc

```diff
@@ -0,0 +1,148 @@
+module [
+    Result,
+    is_ok,
+    is_err,
+    is_eq,
+    map_ok,
+    map_err,
+    map_both,
+    map2,
+    try,
+    on_err,
+    on_err!,
+    with_default,
+]
+
+import Bool exposing [Bool.*]
+
+## The result of an operation that could fail: either the operation went
+## okay, or else there was an error of some sort.
+Result ok err := [Ok ok, Err err]
```

Comment:
I think this should be using parens like `Result(ok, err)` for the RHS type declaration, and `[Ok(ok), Err(err)]` for the LHS annotation